### PR TITLE
WIP: build travis shell scripts for more flexibility

### PR DESCRIPTION
**Pull request by CJ-Wright**
This would allow us to change most of the behaviors of the travis workers by PRing into this repo. Currently one must PR into the workers, which is a pain for standardization.

PRs to the workers will need to go in after this goes in.

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Attn: @isuruf @justcalamari

**Comment by isuruf** [comment]
Can we combine 00 and 01 now that we are making changes?

**Comment by CJ-Wright** [comment]
Sure, I'd like to leave the worker there, we can back fill with 01, 02, or 03 as needed.

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+export START_TIME="$(date +%s)"
```
Comment (CJ-Wright) [inline setup.sh]: The hope is to not time out on Travis

**Comment by CJ-Wright** [comment]
For context

Script | API calls | time
-- | -- | --
00 | few | 4:12
01 | 0 | 21:09
02 | 0 | 7:10
03 | few | 43:43

So we can double up 00-01 and 02-03 and still have more time on each than we had in the past. Most importantly we won't sacrifice time on 03 (although that has been completing without timeout recently, although although that may change with the new code (since there are more things to do)).

At the end of the day this PR will give us more flexibility so if we don't get it right the first time, we're just a single push away from fixing it.

**Pull request merged**